src/julia_threads.h

```diff
@@ -428,8 +442,10 @@ static inline void jl_mutex_lock_nogc(jl_mutex_t *lock)
 
 static inline void jl_mutex_lock(jl_mutex_t *lock)
 {
+    JL_SIGATOMIC_BEGIN();
```

Comment:
True, this is not new though, we have many places with `defer_sigint` set but still allocate and allow finalizers to run. I'm not sure how should the two interact though, in the long term model of running the finalizers on a different thread, it seems like we should run all the finalizers with `defer_sigint` set...